client/src/components/frontend/Resource/Card.js

```diff
@@ -74,12 +75,21 @@ export default class ResourceCard extends Component {
     return text;
   }
 
+  getCollectionResourceId(resource) {
+    if (!resource) return null;
+    const collectionResources = resource.relationships.collectionResources;
+    const out = find(collectionResources, (obj) => {
+      return obj.attributes.collectionId === this.props.context.id;
+    });
+    return out.id;
+  }
+
   detailUrl() {
     const context = this.props.context;
     if (context.type === "collections") {
-      const crid = this.props.resource.id;
       const pid = context.relationships.project.id;
       const cid = context.id;
+      const crid = this.getCollectionResourceId(this.props.resource);
```

Comment:
I had to include the CollectionResources for a Resource because of the different IDs.  Is there a better way we could handle this or does this seem appropriate? It seems like either way we get the resources (by CollectionResource or Resources) we need to reference the association.